### PR TITLE
Add `--cxml` flag

### LINKED ISSUE #15: Add `--xml` flag to structure output for Claude's long context window
# Add `--xml` flag to structure output for Claude's long context window

## Background

Anthropic has provided [specific guidelines](https://docs.anthropic.com/claude/docs/long-context-window-tips) for optimally structuring prompts to take advantage of Claude's extended context window (up to 200K tokens for Claude 3 models).

In particular, they recommend wrapping long input documents in XML tags to clearly delineate the boundaries between the documents and the rest of the prompt. This allows Claude to more accurately process the information.

## Proposal

To better support using files-to-prompt output with Claude, a new `--xml` flag should be added. When this flag is set, the tool should structure its output like this:

```xml
Here are some documents for you to reference for your task:

<documents>
<document path="path/to/file1.txt">
Contents of file1.txt
</document>

<document path="path/to/file2.txt">
Contents of file2.txt
</document>
</documents>
```

The `<documents>` and `<document>` tags clearly separate the file contents from any additional prompt text that may be provided after.

## Additional Considerations

- The `--xml` flag should be optional, preserving the default untagged output for other use as needed
- Documentation and usage examples should be updated to cover the new `--xml` option
- The XML tag output feature should be covered by new unit tests

### PR DESCRIPTION
**Pull request by lexh**
Closes #15

### CODE REVIEW

**Comment by jefftriplett** [comment]
This is great and saves me a ton of time! Thank you for opening the pull request.

**Comment by simonw** [comment]
Yeah let’s fix that and ship this!

File: files_to_prompt/cli.py [hunk 1]
```diff
@@ -114,13 +135,41 @@ def cli(paths, include_hidden, ignore_gitignore, ignore_patterns):
...
+    </documents>
     """
     gitignore_rules = []
     for path in paths:
         if not os.path.exists(path):
             raise click.BadArgumentUsage(f"Path does not exist: {path}")
         if not ignore_gitignore:
             gitignore_rules.extend(read_gitignore(os.path.dirname(path)))
+        if xml and path == paths[0]:
+            click.echo("Here are some documents for you to reference for your task:")
```
Comment (simonw) [inline files_to_prompt/cli.py:161]: I don’t like including this prompt here

File: files_to_prompt/cli.py [hunk 2]
```diff
@@ -114,13 +135,41 @@ def cli(paths, include_hidden, ignore_gitignore, ignore_patterns):
...
+    Here are some documents for you to reference for your task:
+
+    <documents>
+    <document path="path/to/file1.txt">
+    Contents of file1.txt
+    </document>
+
+    <document path="path/to/file2.txt">
+    Contents of file2.txt
+    </document>
```
Comment (jefftriplett) [inline files_to_prompt/cli.py:150]: Just as a head up, this format has drifted or doesn't quite match Claude's latest guidelines. It's a straightforward conversion, though. https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/long-context-tips#example-multi-document-structure

**Comment by simonw** [comment]
The fact that this doesn’t guarantee a valid XML document (that could then be handled by an XML parser) _really_ bugs me!

A few options:

1. Have it produce valid XML, which mainly means applying correct escaping to text between the XML tags. I’m worried this may hurt Claude’s performance, we would need to test that but I’m not sure how best to do that.
2. Admit that this isn’t valid XML and change the option to something like `-c/--cxml` (for Claude XML).
3. Call this XML but have another option, `--valid-xml` - which *does* produce valid XML and by its very existence calls out the Claude not-valid-XML thing.

I’m leaning towards 2.

**Comment by jefftriplett** [comment]
> * Admit that this isn’t valid XML and change the option to something like `-c/--cxml` (for Claude XML).

This feels like a good compromise and opens the door for a valid `--xml` option later. 

I have tested with both valid and Claude's XML, and I don't see any noticeable improvements either way. They have other features where randomly dropping in an xml tag has meaning in an otherwise other text sea of chaos.

File: files_to_prompt/cli.py (hunk 2 above)
Comment (simonw) [reply files_to_prompt/cli.py:150]: See d016523c221ea2169d6bab2b266d034a7430f197

**Pull request merged**